### PR TITLE
Classification consumer

### PR DESCRIPTION
**Pull request by TordJoranger**

### CODE REVIEW

File: src/Equinor.ProCoSys.Completion.Command/MessageProducers/QueueNames.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ﻿namespace Equinor.ProCoSys.Completion.Command.MessageProducers;
 public static class QueueNames
 {
+    public const string ClassificationCompletionTransferQueue = "classificationCompletionTransferQueue";
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.Command/MessageProducers/QueueNames.cs:4]: why const and not static here. Either const for all or static for all

File: src/Equinor.ProCoSys.Completion.Domain/AggregateModels/LibraryAggregate/LibraryItem.cs [hunk 2]
```diff
@@ -42,7 +43,16 @@ public LibraryItem(string plant, Guid guid, string code, string description, Lib
 
     public IReadOnlyCollection<Classification> Classifications => _classifications.AsReadOnly();
 
-    public void AddClassification(Classification classification) => _classifications.Add(classification);
+    public void AddClassification(Classification classification)
+    {
+        if(!ClassificationExistsOrNameInUse(classification))
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.Domain/AggregateModels/LibraryAggregate/LibraryItem.cs:48]: Add unittests, assert that you can add two different classification objects with same Name, but number of c is still 1

File: src/Equinor.ProCoSys.Completion.Infrastructure/EntityConfigurations/ClassificationConfiguration.cs [hunk 3]
```diff
@@ -10,7 +10,9 @@ internal class ClassificationConfiguration : IEntityTypeConfiguration<Classifica
     public void Configure(EntityTypeBuilder<Classification> builder)
     {
         builder.ConfigureSystemVersioning();
-
+        builder.Property(x => x.Guid).ValueGeneratedNever();
+        builder.HasIndex("Name","LibraryItemId").IsUnique();
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.Infrastructure/EntityConfigurations/ClassificationConfiguration.cs:14]: :-)

File: src/Equinor.ProCoSys.Completion.WebApi/Synchronization/ClassificationConsumer.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    public async Task Consume(ConsumeContext<ClassificationEvent> context)
+    {
+        var classification = context.Message;
+        
+        if (classification.Behavior == "delete")
+        {
+           var deleted = await dbContext.Classifications.Where(c => c.Guid == classification.ProCoSysGuid).ExecuteDeleteAsync();
+            logger.LogDebug("Classification delete Message Consumed: {MessageId} \n Guid {Guid} \n Deleted {Deleted} rows",
+                 context.MessageId, classification.ProCoSysGuid, deleted);
+            return;
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.WebApi/Synchronization/ClassificationConsumer.cs:29]: I guess the delete will not be saved?!

File: src/Equinor.ProCoSys.Completion.WebApi/Synchronization/ClassificationConsumer.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+namespace Equinor.ProCoSys.Completion.WebApi.Synchronization;
+
+[UsedImplicitly]
+public class ClassificationConsumer(
+    ILogger<ClassificationConsumer> logger,
+    CompletionContext dbContext) 
+    : IConsumer<ClassificationEvent>
+{
+    public async Task Consume(ConsumeContext<ClassificationEvent> context)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.WebApi/Synchronization/ClassificationConsumer.cs:20]: Add a minimum of unit test, at least one Create and one Delete

**Review by janid1967 — APPROVED** [review]
Pre-approved since I go at vacation, but I like to see unit tests where mentioned on comments ;-)

File: src/Equinor.ProCoSys.Completion.WebApi/Synchronization/ClassificationConsumer.cs (hunk 4 above)
Comment (TordJoranger) [reply src/Equinor.ProCoSys.Completion.WebApi/Synchronization/ClassificationConsumer.cs:29]: .ExecuteDeleteAsync(); does not require save.(more specificaly, it saves on its own, and does not use tracking) 
Bit of a workaround due to not having libraryItemGuid in delete messages. 
alternative would be to create a dummyobject
```
var toDelete = new Classification(busmessage.ProCoSysGuid, "PunchPriority)
dbContext.Classification.Remove(toDelete)
await dbContext.SaveAsync()

```

**Pull request merged**